### PR TITLE
common: Struct ceph_clock

### PR DESCRIPTION
**Pull request by shinobu-x**
Ultimate goal of this PR is to use same interface (wrapper) with same accuracy to get timepoint and latency.

Example use cases in this PR.

Signed-off-by: Shinobu Kinjo <shinobu@redhat.com>

### CODE REVIEW

**Comment by branch-predictor** [comment]
It's quite heavyweight compared to "regular" types (72 bytes of storage for new ```ceph_clock``` vs 8 for ```utime_t``` or ```mono_time```).

File: src/common/ceph_clock.h [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+#include <type_traits>
+#include <time.h>
+
+using namespace std::chrono;
+
+struct ceph_clock {
+  ceph_clock() :
+    mc_start_(ceph::mono_clock::now()),
+    cc_start_(ceph::coarse_mono_clock::now()),
+    ut_start_(ceph_clock_now_()) {}
```
Comment (cbodley) [inline src/common/ceph_clock.h]: i really don't think we want to read from all 3 clocks on construction

**Comment by shinobu-x** [comment]
@cbodley The purpose of this PR is to replace `ceph_clock_now` to `coarse_mono_clock` to get timestamp as much as possible. For this purpose, this PR is intended to provide interfaces for this replacement.

Current code in this PR is not enough sophisticated. So there would be unnecessary lines or lines need to be added for simplifying the replacement.

**Comment by cbodley** [comment]
> The purpose of this PR is to replace ceph_clock_now to coarse_mono_clock to get timestamp as much as possible

is it not enough to just replace calls to `ceph_clock_now()` with `ceph::coarse_mono_clock::now()`? what motivates the extra sophistication?

**Comment by shinobu-x** [comment]
> what motivates the extra sophistication?

ignore this at this time.

File: src/common/ceph_clock.h (hunk 1 above)
Comment (shinobu-x) [reply src/common/ceph_clock.h]: the purpose of distinguishing starting time from construction time is that users of this wrapper should have responsibility of starting point of calculating latency.

we should be accurate as much as possible when it comes to latency.

**Comment by shinobu-x** [comment]
test this please

**Pull request closed without merging**